src/sync/peer-sync-controller.js

```diff
@@ -184,18 +181,45 @@ export class PeerSyncController {
     }
     this.#log('capability %o', this.#syncCapability)
 
-    // If any namespace has new data, update what is enabled
-    if (Object.values(didUpdate).indexOf(true) > -1) {
-      this.#updateEnabledNamespaces()
-    }
+    this.#updateEnabledNamespaces()
   }
 
+  /**
+   * Enable and disable the appropriate namespaces.
+   *
+   * If replicating no namespace groups, all namespaces are disabled.
+   *
+   * If only replicating the initial namespace groups, only the initial
+   * namespaces are replicated, assuming the capability permits.
+   *
+   * If replicating all namespaces, everything is replicated. However, data
+   * namespaces are only enabled after the initial namespaces have synced. And
+   * again, capabilities are checked.
+   */
   #updateEnabledNamespaces() {
-    // - If the sync capability is unknown, then the namespace is disabled,
-    //   apart from the auth namespace.
-    // - If sync capability is allowed, the "pre-sync" namespaces are enabled,
-    //   and if data sync is enabled, then all namespaces are enabled
+    /** @type {boolean} */ let isAnySyncEnabled
+    /** @type {boolean} */ let isDataSyncEnabled
+    switch (this.#namespaceGroupToReplicate) {
+      case 'none':
+        isAnySyncEnabled = isDataSyncEnabled = false
+        break
+      case 'initial only':
+        isAnySyncEnabled = true
+        isDataSyncEnabled = false
+        break
+      case 'all':
+        isAnySyncEnabled = isDataSyncEnabled = true
+        break
+      default:
+        throw new ExhaustivenessError(this.#namespaceGroupToReplicate)
+    }
+
```

Comment:
Personal preference comment: I'm not sure if the indirection created here is helpful, or just makes the code below harder to follow vs. replacing `if (!isAnySyncEnabled)` with `if (this.#nameSpaceGroupToReplicat !== 'none')` and replacing `else if (isDataSyncEnabled)` with `else if (this.#nameSpaceGroupToReplicate === 'all'`?